File: java/clock/src/main/java/Clock.java
public class Clock {
  static final int MINUTE_IN_DAY = 24 * 60;

  int totalMinute;

  Clock(int hour, int minute) {
    totalMinute = Math.floorMod(hour * 60 + minute, MINUTE_IN_DAY);
  }

  void add(int minutesToAdd) {
    totalMinute = Math.floorMod(totalMinute + minutesToAdd, MINUTE_IN_DAY);
  }

  @Override
  public String toString() {
    return String.format("%02d:%02d", totalMinute / 60, totalMinute % 60);
  }

  @Override
  public int hashCode() {
    return totalMinute;
  }

  @Override
  public boolean equals(Object obj) {
    Clock other = (Clock) obj;

    return totalMinute == other.totalMinute;
  }
}

File: python/clock/clock.py
class Clock(object):
    def __init__(self, hour, minute):
        self.total_minutes = (hour * 60 + minute) % (24 * 60)

    def __add__(self, other):
        return Clock(0, self.total_minutes + other)

    def __str__(self):
        return '{:0>2}:{:0>2}'.format(self.total_minutes // 60, self.total_minutes % 60)

    def __eq__(self, other):
        return self.total_minutes == other.total_minutes
